Bug creating Azure Service Fabric Managed Cluster with Authentication Block

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.6

### AzureRM Provider Version

2.90 and latest

### Affected Resource(s)/Data Source(s)

azurerm_service_fabric_managed_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_service_fabric_managed_cluster" "sfc_mng_rg" {
  name                   = "mng-sfc-templatetf"
  resource_group_name    = azurerm_resource_group.sfc_mng_rg.name
  location               = azurerm_resource_group.sfc_mng_rg.location
  http_gateway_port      = 19080
  upgrade_wave           = "Wave2"
  username               = "superuser"
  password               = "superpassword"
  sku                    = "Standard"
  backup_service_enabled = true
  dns_service_enabled    = true


  authentication {
    active_directory {
      client_application_id  = "xxxxxxxxxxxxxxxxxxxxxxxxxx"
      cluster_application_id = "xxxxxxxxxxxxxxxxxxxxxxx"
      tenant_id              = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
    }

    certificate {
      thumbprint = "certthumbprint"
      type       = "AdminClient"
    }
  }


  lb_rule {
    backend_port   = 443
    frontend_port  = 443
    probe_protocol = "tcp"
    protocol       = "tcp"
  }


  client_connection_port = 19000

  node_type {
    data_disk_size_gb      = 64
    name                   = "NT1"
    primary                = true
    application_port_range = "20000-30000"
    ephemeral_port_range   = "49152-65534"

    vm_size            = "Standard_D2s_v3"
    vm_image_publisher = "MicrosoftWindowsServer"
    vm_image_sku       = "2019-Datacenter-with-Containers"
    vm_image_offer     = "WindowsServer"
    vm_image_version   = "latest"
    vm_instance_count  = 5
  }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v2.90.0_x5.exe plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 126 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/servicefabricmanaged.flattenClusterProperties(0xc001d16370)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go:579 +0xa2c
github.com/hashicorp/terraform-provider-azurerm/internal/services/servicefabricmanaged.ClusterResource.Read.func1({0x66e5ce8, 0xc001cc0a80}, {0xc001479880, {0x66e7338, 0xc00010b380}, 0xc001871f00, 0x0, {0x66e8f00, 0xaca3a40}})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go:275 +0x4d3
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x66e5ce8, 0xc001cc0a80}, 0x4e94914f000, {0x57146c0, 0xc001479880})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:59 +0x1e6
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x66e5ce8, 0xc001cc0a80}, 0xc001cf76c0, {0x57146c0, 0xc001479880})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:168 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0000bd960, {0x66e5c78, 0xc001c8f240}, 0x2, {0x57146c0, 0xc001479880})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:341 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000bd960, {0x66e5c78, 0xc001c8f240}, 0xc001c21ad0, 0xc001871d80, {0x57146c0, 0xc001479880})     
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0006541c8, {0x66e5c78, 0xc001c8f240}, 0xc0004a54f0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000310700, {0x66e5d20, 0xc001cb1320}, 0xc000c97810)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5c49000, 0xc000310700}, {0x66e5d20, 0xc001cb1320}, 0xc001cc03c0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000132fc0, {0x6734388, 0xc00043f680}, 0xc001cbb200, 0xc000e3a7b0, 0xac3e2c0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000132fc0, {0x6734388, 0xc00043f680}, 0xc001cbb200, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.90.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Finish the cluster deployment.

### Actual Behaviour

Crash.

### Steps to Reproduce

Try to deploy the Managed SFC with the Authentication block, if the Authentication block is not in the code the issues don't happen.

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go
+++ b/internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go
@@ -679,6 +679,7 @@ func flattenClusterProperties(cluster *managedcluster.ManagedCluster) *ClusterRe
 	model.Username = properties.AdminUserName
 
 	if aad := properties.AzureActiveDirectory; aad != nil {
+		model.Authentication = append(model.Authentication, Authentication{})
 		adModel := ADAuthentication{}
 		adModel.ClientApp = utils.NormalizeNilableString(aad.ClientApplication)
 		adModel.ClusterApp = utils.NormalizeNilableString(aad.ClusterApplication)
@@ -698,6 +699,9 @@ func flattenClusterProperties(cluster *managedcluster.ManagedCluster) *ClusterRe
 				CommonName:      utils.NormalizeNilableString(client.CommonName),
 				Thumbprint:      utils.NormalizeNilableString(client.Thumbprint),
 			}
+		}
+		if len(model.Authentication) == 0 {
+			model.Authentication = append(model.Authentication, Authentication{})
 		}
 		model.Authentication[0].CertAuthentication = certs
 	}

